packages/axios-extension/src/abap/ui5-abap-repository-service.ts

```diff
@@ -99,18 +99,21 @@ function encodeXmlValue(xmlValue: string): string {
  */
 export class Ui5AbapRepositoryService extends ODataService {
     public static readonly PATH = '/sap/opu/odata/UI5/ABAP_REPOSITORY_SRV';
+    private readonly publicUrl: string;
 
     /**
      * Extension of the base constructor to set preferred response format if not provided by caller.
      *
      * @param config optional base configuration for Axios
+     * @param publicUrl optional string representing the host exposing the ABAP repository
      */
-    public constructor(config?: AxiosRequestConfig) {
+    public constructor(config?: AxiosRequestConfig, publicUrl?: string) {
```

Comment:
Could we keep it in one object in case we want to add more in the future

```
public constructor(config?: AxiosRequestConfig & { publicUrl?: string }) {
```